spec/commands_generator_spec.rb

```diff
@@ -0,0 +1,9 @@
+require "gym/commands_generator"
+
+describe Gym do
+  describe Gym::CommandsGenerator do
+    it "validates available_options" do
+      # this runs the initialization code in the commands generator and detects invalid options
```

Comment:
Are you sure this will fail if there is a duplicate entry? You don't initialise anything.